game-core/src/main/java/games/strategy/triplea/util/UnitSeparator.java

```diff
@@ -19,11 +19,33 @@
 import java.util.Map;
 import java.util.Set;
 import java.util.TreeSet;
+import javax.annotation.Nullable;
+import lombok.Builder;
 
 /** Separates a group of units into distinct categories. */
 public class UnitSeparator {
   private UnitSeparator() {}
 
+  @Builder(toBuilder = true)
+  public static class SeparatorCategories {
+    /**
+     * if not null, then will group units with the same dependents (compares type, owner, and
+     * amount)
+     */
+    @Builder.Default @Nullable final Map<Unit, Collection<Unit>> dependents = null;
+    /** whether to categorize by movement */
+    @Builder.Default final boolean movement = false;
+    /** whether to categorize by transport cost */
+    @Builder.Default final boolean transportCost = false;
+    /** whether to categorize transports by movement */
+    @Builder.Default final boolean transportMovement = false;
+    /**
+     * If true then sort the categories in UnitCategory order; if false, then leave categories in
```

Comment:
Consider defining 'UnitCategory (sort) order', I'm not 100% clear what it really means. Secondarily the 'sort' variable perhaps could be given some added context (renamed) to help convey more precisely what the sort is. Perhaps the javadoc should also mention when it would be desirable to sort. Third, 'original order' could perhaps be more precise, I would presume that is the order units are added to the separator (but perhaps it is something else, having the documentation be more exact would clarify this).

Reply:
After going through the commit history, I determined that the sort is no longer needed.  So it is removed in https://github.com/triplea-game/triplea/pull/8920/commits/e0d2a13d3a57c8bc7e8244dc134c99f642bb8706.